### PR TITLE
Python: __bool__ does not raise TypeError by default

### PR DESCRIPTION
**Pull request by yoff**
This PR just adds a test to highlight the problem. I will later add a commit to it to also fix the problem.

### CODE REVIEW

File: python/ql/test/query-tests/Functions/general/protocols.py [hunk 1]
```diff
@@ -116,3 +116,6 @@ class OK(object):
     def __call__(self):
         yield 0
         raise StopIteration
+
+    def __bool__(self):
+        raise TypeError
```
Comment (RasmusWL) [inline python/ql/test/query-tests/Functions/general/protocols.py:121]:
```suggestion
    # __bool__ returns `True` by default, so raising `TypeError` should not give an alert
    # FP reported in https://github.com/github/codeql/issues/2388
    def __bool__(self):
        raise TypeError
```

**Comment by yoff** [comment]
Thanks, just to make sure I understand: Between step 1 and 2, the `.expected`-file does not actually reflect what we know of the world? That feels a bit wrong :-) Is it not possible to see the intermediate failings?

**Comment by yoff** [comment]
Ah, Taus has explained this in more detail in my onboarding issue.

**Comment by yoff** [comment]
That is a good point, I suppose that should be considered part of solving the problem :-)

**Comment by yoff** [comment]
Ah, now I see why the merge was needed. It is because I committed your suggested change. I should've just pulled right after that.

File: python/ql/test/query-tests/Functions/general/protocols.py [hunk 2]
```diff
@@ -118,4 +118,8 @@ def __call__(self):
         raise StopIteration
 
     def __bool__(self):
-        raise TypeError
\ No newline at end of file
+        raise TypeError
+
+class BadBool(object):
+    def __bool__(self):
+        raise ZeroDivisionError()
```
Comment (RasmusWL) [inline python/ql/test/query-tests/Functions/general/protocols.py]: nitpick: could we move this to `IncorrectSpecialMethods`?

**Comment by yoff** [comment]
redo failed

**Comment by yoff** [comment]
Check failed due to incorrect line numbers. I must have not run the tests after changing comments or something.

File: python/ql/test/query-tests/Functions/general/protocols.py (hunk 2 above)
Comment (yoff) [reply python/ql/test/query-tests/Functions/general/protocols.py]: Yes! Will do.

**Pull request merged**